api/app/routers/hfi_calc.py

```diff
@@ -40,18 +42,21 @@
 def get_prepared_request(
         session: Session,
         fire_centre_id: int,
-        start_date: Optional[date],
-        end_date: Optional[date] = None) -> Tuple[HFIResultRequest, bool, List[FireStartRange]]:
+        date_range: Optional[DateRange]) -> Tuple[HFIResultRequest, bool, List[FireStartRange]]:
     """ Attempt to load the most recent request from the database, failing that creates a new request all
     set up with default values.
 
     TODO: give this function a better name.
     """
     fire_centre_fire_start_ranges = list(load_fire_start_ranges(session, fire_centre_id))
-    date_range = DateRange(start_date=start_date, end_date=end_date)
-    stored_request = get_most_recent_updated_hfi_request(session,
-                                                         fire_centre_id,
-                                                         date_range.start_date)
+    if date_range:
+        stored_request = get_most_recent_updated_hfi_request(session,
+                                                             fire_centre_id,
+                                                             date_range)
+        # NOTE: We could be real nice here, and look for a prep period that intercepts, and grab data there.
+    else:
+        # No date range specified!
+        stored_request = get_more_recent_updated_hfi_request_in_range(session, fire_centre_id)
```

Comment:
Shouldn't this be the other way around? If there's no date range how would we look up a request in a range?

Reply:
Bad function name - I've renamed it `get_more_recent_updated_hfi_request_for_current_date` ; So it just looks at the server date to figure out the relevant request.